Sources/SimplenoteFoundation/Extensions/String+RangeConversion.swift

```diff
@@ -0,0 +1,44 @@
+import Foundation
+
+
+// MARK: - String RangeConversion API(s0
+//
+extension String {
+
+    /// Returns the receiver as a Foundation String
+    ///
+    public var nsString: NSString {
+        self as NSString
+    }
+
+    /// Converts a UTF16 Location  into a String.Index
+    ///
+    public func indexFromLocation(_ location: Int) -> String.Index? {
+        guard let unicodeLocation = utf16.index(utf16.startIndex, offsetBy: location, limitedBy: utf16.endIndex),
+            let location = unicodeLocation.samePosition(in: self) else {
+                return nil
+        }
+
+        return location
+    }
+
+    /// Maps a `String.Index` in the receiver's coordinates into coordinates valid for the specified Substring, in the target Range
+    ///
+    public func transportIndex(_ index: String.Index, to range: Range<String.Index>, in substring: String) -> String.Index? {
+        let locationInSubstring = distance(from: startIndex, to: index) - distance(from: startIndex, to: range.lowerBound)
+        return substring.index(substring.startIndex, offsetBy: locationInSubstring, limitedBy: substring.endIndex)
+    }
+
+    /// Converts a `Range<String.Index>` into an UTF16 NSRange.
+    ///
+    public func utf16NSRange(from range: Range<String.Index>) -> NSRange {
+        guard let lowerBound = range.lowerBound.samePosition(in: utf16), let upperBound = range.upperBound.samePosition(in: utf16) else {
+            fatalError()
```

Comment:
Random thought: 
If you wanted to avoid a `fatalError` would returning `NSMakeRange(NSNotFound,NSNotFound)`, or something like it, be an option?

Reply:
Switching over `assertionFailure` + NSNotFound. Thank you!